lib/filterTeachersByAvailability.js

```diff
@@ -0,0 +1,21 @@
+module.exports = (studentAvailability, teachersArray) =>
+  teachersArray.filter(teacher => {
+    let include;
```

Comment:
There may be a better way to do this filtering, but this can work as a good way to start. I would recommend a slightly more concise way of filtering though, where you initialize `include` as `false`, set it to `true` conditionally, and then just return `include` at the end. So a kind of pseudo code representation would be:
```
teachersArray.filter(teacher => {
  let include = false;
  teacher.availability.map(dayAvailable => {
    if (condition met) {
      let = true;
    }
  });
  return include;
});
```

Reply:
Yes, that is much more concise and straightforward.  I would be interested in knowing a better way of filtering here to improve my skills as a developer.